src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2153,6 +2153,8 @@ NetworkOPsImp::pubValidation(std::shared_ptr<STValidation> const& val)
         jvObj[jss::flags] = val->getFlags();
         jvObj[jss::signing_time] = *(*val)[~sfSigningTime];
         jvObj[jss::data] = strHex(val->getSerializer().slice());
+        if (auto const netid = app_.overlay().networkID())
```

Comment:
Is there a situation where there is no network ID?

Reply:
Yeah, that occurs here: https://github.com/XRPLF/rippled/blob/fad9d639bf52e727aa9cc1bedbdcdb49aa4d36de/src/ripple/overlay/impl/OverlayImpl.cpp#L1617

But if no network_id is specified, how will a validator know which peers it must connect with?

Reply:
I think the ideal would be for nodes for nodes on different networks to not connect to each other in the first place. And if they are connected, they should disconnect from each other. That should automatically solve the issue since you wouldn't get validations from other networks to begin with.

Reply:
My understanding that would be done when choosing which peers to use and add? @mvadari @ckeshava is that correct. 

It also has a dependency on the id being present meaning all nodes need to update to a rippled version publishing this information with its peers information. Only then would it be possible to then exclude peers not on your network.

That fine if that is the chosen route to achieve that but it would need to be split across a few releases.

Reply:
I don't know much about this topic. 
We are validating the `network_id` inside the transaction engine, but we are not validating it inside the classes `Peer`, `Overlay` or whilst setting up the UNL. Fixing the gap in these classes should fix the chatter in the gossip protocol.

I don't foresee any breaking of functionality if this idea were to be implemented immediately. The cross-communication of validations across different networks occurs due to mis-configurations. Yes, I believe it presents more work for the validator operator.

Reply:
im going to leave that off to another request as I also don't fully understand how best to achieve that.